### ISSUE TITLE
Twitch in-stream advertisements not being blocked

### ISSUE THREAD

**Issue by UndercoverNL** [brave/brave-browser#22370]
## Description 
When watching twitch.tv streams none of the ads playing in stream are blocked, both the pre-roll and mid-stream ads are just being played like on another browser.

## Steps to Reproduce
Watch a stream on https://twitch.tv/, or swich to a few streams and at some of them a pre-roll ad will come-up

## Actual result:
All kind of in-stream video ads being played

## Expected result:
No (video) ads being played before or in streams

## Reproduces how often: 
Always an ad is being played by the streamer / twitch

## Brave version (brave://version info)
1.37.116 Chromium: 100.0.4896.127 (Official Build) (64-bit) 

## Version/Channel Information:
It also happens in Brave Beta, so I guess it happens on all versions

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Idk what this is

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?  no
- Does the issue resolve itself when disabling Brave Rewards?  no
- Is the issue reproducible on the latest version of Chrome? yes

**Comment by UndercoverNL** [comment]
I don't see why its related to 23014, it should be blocked by Brave it self, not by using a third party app.

Anyway, it is still an problem which should be an higher priority to me. A lot of people use twitch and Brave should filter those ads out too.

**Comment by Zerophase** [comment]
Trying aggresively blocking ads to see if it works.

**Comment by Zerophase** [comment]
The aggressively block ads and trackers option.

**Comment by ryanbr** [comment]
With the latest https://github.com/pixeltris/TwitchAdSolutions

Testing the top 10 twitch extensions with twitch adblocking in Brave.

| Extension  | Breaks Brave? |  Extension works? | Brave still blocks ads? |  Comments |
| ------------- | -------------  | ------------- | ------------- | ------------- |
|[ Twitch Lurker](https://chromewebstore.google.com/detail/twitch-lurker/fkjghajhfjamfjcmdkbangbeogbagnjf)  | To be tested  |   To be tested
| [Twitch High Quality](https://chromewebstore.google.com/detail/twitch-high-quality/khmgdjbkacbgpcaoenlibdpokgelffkg)  | To be tested  | To be tested
| [BetterTTV](https://chromewebstore.google.com/detail/betterttv/ajopnjidmegmdimjlfnijceegpefgped)  | To be tested  |   To be tested
| [Alternate Player for Twitch.tv](https://chromewebstore.google.com/detail/alternate-player-for-twit/bhplkbgoehhhddaoolmakpocnenplmhf)  | To be tested  |  To be tested
| [Unwanted Twitch](https://chromewebstore.google.com/detail/unwanted-twitch/egbpddkgpjmliolmpjenjomflclekjld)  | To be tested  |  To be tested
| [Previews (For TTV & YT](https://chromewebstore.google.com/detail/previews-for-ttv-yt/hpmbiinljekjjcjgijnlbmgcmoonclah)  | To be tested  |  To be tested
| [Better Twitch Adblock](https://chromewebstore.google.com/detail/better-twitch-adblock/bfeoploanlacfeohdakiecicjnjcooji)  | To be tested  |  To be tested
| [Stream Cleaner](https://chromewebstore.google.com/detail/stream-cleaner/lehcglgkjkamolcflammloedahjocbbg)  | To be tested  | To be tested
| [Adblock for Twitch](https://chromewebstore.google.com/detail/adblock-for-twitch/mdomkpjejpboocpojfikalapgholajdc)  | To be tested  |  To be tested
| [TTV LOL PRO](https://chromewebstore.google.com/detail/ttv-lol-pro/bpaoeijjlplfjbagceilcgbkcdjbomjd)  | To be tested  | To be tested

**Comment by ryanbr** [comment]
### Update 25th Jan/2025

TLDR; No bad interactions with Brave Twitch adblocking and these popular extensions, no black screens. Video playback was fine.

| Brave twitch adblocker  |  Extension  |
| -------------------------- | ------------|
| No |                         TTV Lol Pro 2.3.10 adblocker working |
| Yes |                         Stream cleaner 1.17.0  (not working?) |
| Yes |                         Better Twitch AdBlock (3.0.3)  |
| Yes |                         Adblock for Twitch (1.0.2) |
| Yes |                         FrankerFaceZ (4.76.3.0) + 7TV (3.0.9) |
| No |                         (blocking ads still) ALL Extensions enabled. (1) |
| Yes |                        ALL Extensions enabled except TTV Lol pro (1) (2). |

1. Unsure if running all extensions would cause any issues, bad interactions expected. But not Brave related.
2. Disabled TTV Lol Pro since it has its own adblock

**Comment by ryanbr** [comment]
As of (2 Feb) + Experimental:

- **Android Tabelet:**

![Image](https://github.com/user-attachments/assets/af26b6eb-7194-4e81-872f-1208aab9077b)

- **IOS/iPAD **

![Image](https://github.com/user-attachments/assets/f310286f-4e53-49a6-8451-d6b80445fce5)

**Comment by Zerophase** [comment]
I have not been seeing them since. But, it might also be my firewall blocking ads through Sensei, Unbound, and me connecting to Twitch from Hong Kong.
